magma/when.py

```diff
@@ -143,17 +131,27 @@ def _add_reg_enables(self):
         is unnecessary and promote the idiom of not explicitly wiring enables.
         """
         # Copy with list because this can add conditional wires
-        for wire in list(self._conditional_wires):
+        _conditional_wires = list(self._conditional_wires)
+
+        # Only wire enables once otherwise an elaborated register input value
+        # may trigger multiple wirings
+        # NOTE(leonardt): If we wanted to allow users to wire explicitly, we
+        # could lookup whether a wire to enable exists in _conditional_wires
+        # (which would also avoid multiple wirings)
+        seen = set()
```

Comment:
When running into this seen issue, I realized there is a simple way to allow users to explicitly wire enable if we wanted, but it still seems odd to support it (maybe they want to choose the wired value without updating for a case?)

Reply:
(Alternatively, we may decide to force the user to use the Enable type, and it's a simple update to the user code